File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        set<int> st;
        int n = nums.size();
        for(int i=0; i<n; i++){
            if(st.find(nums[i]) != st.end())
                return nums[i];
            else
                st.insert(nums[i]);
        }
        return -1;
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int ind = nums[0];

        while (ind != -1) {
            if (nums[ind] == -1)
                return ind;

            int temp = nums[ind];
            nums[ind] = -1;
            ind = temp;
        }

        return 0;
    }
}
